CMakeLists.txt

```diff
@@ -359,7 +359,9 @@ if (ENABLE_EXTRA_WARNINGS)
     -Wmissing-format-attribute
     -Wold-style-definition
     -Wundef
-    -Wignored-qualifiers)
+    -Wignored-qualifiers
+    -Wmissing-declarations
+    -Winline )
```

Comment:
I think `-Winline` is something that we do not really need.
* clang just accepts and ignores this option
* gcc provide some diagnostics when a function is marked inline, but it cannot be inlined because of some C feature (such as long jump, alloca, etc...).

As the compiler free to ignore the `inline` by default, such a warning imho is not that useful. 

For details see: https://gcc.gnu.org/onlinedocs/gcc/Inline.html

Reply:
Hey @Kokan 
I thought of fixing multiple declarations along with inline. But there are about 168 instances of such warning. How do I go about this PR?

Reply:
One step at a time :)

First of all I think understanding the warning is important. What kind of issues does it prevent, and what benefit it has. Once you get familiar with that, and still decided to proceed with fixing those warnings, as a side effect you probably also have some idea how to fix such an issue. (you should have). This could provide you a nice template that is not hard to do but time consuming due to 168 of them. 

So here goes the fixing mostly sameish issues with a chainsaw, and wish for the best not to bump into a not possible or very hard to resolve type of situation.
At that point you could share the results with us (but please keep your branch up to date for the master), so we could merge and both be happy without those nasty warnings.

Disclaimer: I have just a rough idea what this `-Wignored-qualifiers` warning is about, and I am lazy enough not to read more about it. But marking some function `static` made a few of those warning go away (not saying it is a solution but might).